outputs/cloudevents.go

```diff
@@ -0,0 +1,66 @@
+package outputs
+
+import (
+	"context"
+	"log"
+	"strings"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"github.com/falcosecurity/falcosidekick/types"
+)
+
+// CloudEvent produces a CloudEvent and sends to the CloudEvents consumers.
+func (c *Client) CloudEvents(falcopayload types.FalcoPayload) {
+	c.Stats.CloudEvents.Add(Total, 1)
+
+	if c.ce == nil {
+		client, err := cloudevents.NewDefaultClient()
+		if err != nil {
+			go c.CountMetric(Outputs, 1, []string{"output:cloudevents", "status:error"})
+			log.Println("[ERROR] : CloudEvents - NewDefaultClient", err)
+			return
+		}
+		c.ce = client
+	}
+
+	ctx := cloudevents.ContextWithTarget(context.Background(), c.EndpointURL.String())
+
+	event := cloudevents.NewEvent()
+	event.SetTime(falcopayload.Time)
+	event.SetSource("falco.org") // TODO: this should have some info on the falco server that made the event.
+	event.SetType(toEventType(falcopayload.Rule))
+	event.SetExtension("priority", falcopayload.Priority)
+	event.SetExtension("rule", falcopayload.Rule)
+
+	if err := event.SetData(cloudevents.ApplicationJSON, falcopayload); err != nil {
+		log.Println("[ERROR] : CloudEvents, failed to set data", err)
+	}
+
+	if result := c.ce.Send(ctx, event); cloudevents.IsUndelivered(result) {
+		go c.CountMetric(Outputs, 1, []string{"output:cloudevents", "status:error"})
+		c.Stats.Webhook.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "cloudevents", "status": Error}).Inc()
+		log.Println("[ERROR] : CloudEvents -", result)
+		return
+	}
+
+	// Setting the success status
+	go c.CountMetric(Outputs, 1, []string{"output:cloudevents", "status:ok"})
+	c.Stats.Webhook.Add(OK, 1)
+	c.PromStats.Outputs.With(map[string]string{"destination": "cloudevents", "status": OK}).Inc()
+	log.Println("[INFO] : CloudEvents - Send OK")
+}
+
+func toEventType(rule string) string {
+	rule = strings.ToLower(rule)
+
+	// TODO: this could be improved but I do not know what is possible, I am assuming it is a free form entry.
+	for _, old := range []string{".", ":", ","} {
```

Comment:
You may reuse this pattern : https://github.com/falcosecurity/falcosidekick/blob/0403fbd025724f4eb9ea273fbc5f06a277296b23/outputs/nats.go#L28

Reply:
I ended up deciding the event type can be the same, and the rule name is added as an extension